### DISCUSSION TITLE
useEffect is not working on latest version

### DISCUSSION THREAD

**Discussion by Shofol** [vercel/next.js#55931] [App Router] [answered]
I have started a fresh new project using `npx create-next-app@latest` command. Now, I am trying to use `useEffect` hook on `page.tsx` but it's not firing. 

Here's the code-

``` 
"use client";

import Image from "next/image";
import { useEffect } from "react";

export default function Home() {
  useEffect(() => {
    alert("asd");
    console.log("first");
  }, []);

  return (
   // rest of the code
  );
}
```
You can check the repo too- https://github.com/Shofol/nextjs-issue/blob/main/app/page.tsx

**Comment by icyJoseph** [comment]
What Node.js version do you have installed? Also what Next.js version exactly/

**Reply by Shofol** [reply]
16.13.1

**Reply by icyJoseph** [reply, marked as answer]
You should use 16.14 and later:

- https://github.com/vercel/next.js/issues/54269#issuecomment-1685109739
- Docs: https://nextjs.org/docs/app/building-your-application/upgrading/app-router-migration#nodejs-version

**Reply by Shofol** [reply]
Hmm. It's working. It's really weird issue to track. Many thanks for the reference @icyJoseph. Was stuck for a long time!

**Reply by icyJoseph** [reply]
Yeah, it took me some hours to figure it out too... and that's not the only time something stream related, has been a headache these days 😅
